pkg/controllermanager/bootstrappers/bootstrap.go

```diff
@@ -56,10 +94,174 @@ func (b *Bootstrapper) Start(parentCtx context.Context) error {
 		return fmt.Errorf("failed cleaning up no longer required secrets: %w", err)
 	}
 
+	if err := b.bumpProjectResourceQuotas(ctx); err != nil {
+		return fmt.Errorf("failed bumping project resource quotas: %w", err)
+	}
+
 	b.Log.Info("Successfully bootstrapped Garden cluster")
 	return nil
 }
 
+func (b *Bootstrapper) getProjectNamespaces(ctx context.Context) (*metav1.PartialObjectMetadataList, error) {
+	projectNamespaces := &metav1.PartialObjectMetadataList{}
+	projectNamespaces.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Namespace"))
+	if err := b.Client.List(ctx, projectNamespaces, client.MatchingLabels{v1beta1constants.GardenRole: v1beta1constants.GardenRoleProject}); err != nil {
+		return nil, fmt.Errorf("failed listing project namespaces: %w", err)
+	}
+	return projectNamespaces, nil
+}
+
+func (b *Bootstrapper) getResourceQuotas(ctx context.Context, namespace string) (*corev1.ResourceQuotaList, error) {
+	resourceQuotas := &corev1.ResourceQuotaList{}
+	if err := b.Client.List(ctx, resourceQuotas, client.InNamespace(namespace)); err != nil {
+		if !errors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed listing project resource quotas: %w", err)
+		}
+	}
+	return resourceQuotas, nil
+}
+
+func (b *Bootstrapper) getMaximumShootsInProject(ctx context.Context, resourceQuota corev1.ResourceQuota, projectNamespace string) (int64, error) {
+	var maximum int64
+	if limit, hasQuota := resourceQuota.Spec.Hard[corev1.ResourceName(shootCountResource)]; hasQuota {
+		maximum = limit.Value()
+	} else {
+		shootList := &metav1.PartialObjectMetadataList{}
+		shootList.SetGroupVersionKind(gardencorev1beta1.SchemeGroupVersion.WithKind("ShootList"))
+		if err := b.Client.List(ctx, shootList, client.InNamespace(projectNamespace)); err != nil {
+			return 0, fmt.Errorf("could not list shoots in project namespace %q: %w", projectNamespace, err)
+		}
+		maximum = int64(len(shootList.Items))
+	}
+	return maximum, nil
+}
```

Comment:
opinionated: I'd prefer to make use of early returns (and possibly also avoiding the nesting in the else clause).
```suggestion
func (b *Bootstrapper) getMaximumShootsInProject(ctx context.Context, resourceQuota corev1.ResourceQuota, projectNamespace string) (int64, error) {
	if limit, hasQuota := resourceQuota.Spec.Hard[corev1.ResourceName(shootCountResource)]; hasQuota {
		return limit.Value(), nil
	}
	shootList := &metav1.PartialObjectMetadataList{}
	shootList.SetGroupVersionKind(gardencorev1beta1.SchemeGroupVersion.WithKind("ShootList"))
	if err := b.Client.List(ctx, shootList, client.InNamespace(projectNamespace)); err != nil {
		return 0, fmt.Errorf("could not list shoots in project namespace %q: %w", projectNamespace, err)
	}
	return int64(len(shootList.Items)), nil
}
```